src/main/java/ganymedes01/etfuturum/blocks/BlockBarrel.java

```diff
@@ -23,24 +25,35 @@
 import net.minecraft.util.Facing;
 import net.minecraft.util.IIcon;
 import net.minecraft.world.World;
+import org.apache.commons.lang3.ArrayUtils;
 
 import java.util.Random;
 
 public class BlockBarrel extends BlockContainer {
+	public BlockBarrel(){this(TileEntityBarrel.BarrelType.VANILLA);}
+
+	public BlockBarrel(TileEntityBarrel.BarrelType type) {
+		super(type == TileEntityBarrel.BarrelType.VANILLA ? Material.wood : Material.iron);
+		this.type = type;
+		if (type == TileEntityBarrel.BarrelType.VANILLA){
+			this.setStepSound(soundTypeWood);
+			this.setHarvestLevel("axe", 0);
+			this.setBlockName(Utils.getUnlocalisedName("barrel"));
+		}else {
+			this.setStepSound(soundTypeMetal);
+			this.setHarvestLevel("pickaxe", 1);
+			this.setBlockName(Utils.getUnlocalisedName(type.name().toLowerCase()+"_barrel"));
```

Comment:
Set the texture name here rather than building it in `registerBlockIcons`.
```suggestion
			this.setBlockTextureName("barrel");
		} else {
			this.setStepSound(soundTypeMetal);
			this.setHarvestLevel("pickaxe", 1);
			this.setBlockName(Utils.getUnlocalisedName(type.name().toLowerCase() + "_barrel"));
			this.setBlockTextureName("metalbarrels:" + type.name().toLowerCase() + "_barrel");
```